backtracking/power_sum.py

```diff
@@ -0,0 +1,82 @@
+"""
+Problem source: https://www.hackerrank.com/challenges/the-power-sum/problem
+Find the number of ways that a given integer X, can be expressed as the sum
+of the Nth powers of unique, natural numbers. For example, if X=13 and N=2.
+We have to find all combinations of unique squares adding up to 13.
+The only solution is 2^2+3^2.
+"""
+
+from math import pow
+
+sum_ = 0
+count_ = 0
+
+
+def backtrack(x: int, n: int, i: int) -> None:
+    """
+    Backtracking function to find all the possible combinations
+    of powers of natural numbers that add up to x.
+    Parameters:
+        x: The number to be expressed as the sum of
+        the nth powers of unique, natural numbers.
+        n: The power of the natural numbers.
+        i: The current natural number.
+
+    Returns:
+        None
```

Comment:
Don't need to give `Parameters` and `Returns` explanation in here, provide Doctests and their output.

Reply:
Removed